libs/sdk-ui/src/base/measureTitles/ignoreTitlesForSimpleMeasures.ts

```diff
@@ -22,6 +27,25 @@ import {
  * @internal
  */
 export function ignoreTitlesForSimpleMeasures<T extends IInsightDefinition>(insight: T): T {
+    if (
+        insightHasItem(insight, (bucketItem: IAttributeOrMeasure): boolean => {
+            if (!isMeasure(bucketItem)) {
+                return false;
+            }
+            const definition: IMeasureDefinitionType = bucketItem.measure.definition;
+            if (!isMeasureDefinition(definition)) {
+                return false;
+            }
+            return (
+                Array.isArray(definition.measureDefinition.filters) &&
+                definition.measureDefinition.filters.some(isDateFilter)
+            );
+        })
```

Comment:
hmm, i'm thinking that we already have most of the methods to achieve this: `insightMeasures` and `bucketsMeasures`. insightMeasures does not allow to specify predicate but that can be done fastly and in backward compatible fashion. while doing that, please add the predicate also for insightAttributes. I don't know why we haven't added the predicates here.. they make complete sense to have as optional arguments.

you could then rewrite this to something like:

const measures = insightMeasures(insight, (measure) => {
    return measureFilters(measure)?.some(isDateFilter) ?? false;
});

if (measures.length > 0) {
...
}

i suggest to rewrite this PR and get rid of the *HasItem functions; so far we have not needed these functions because code doing lookups always knows whether it is looking for measure or attribute.

Reply:
Thanks, I've updated the PR.